worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,373 @@
+                         "World 4 Gate", "World 5 Gate", "World 6 Gate"]
+
+
+        excluded_items.add(starting_gate[self.options.starting_world])
+
+        if not self.options.shuffle_midrings:
+            excluded_items.add('Middle Ring')
+
+        if not self.options.add_secretlens:
+            excluded_items.add('Secret Lens')
+
+        if not self.options.extras_enabled:
+            excluded_items.add('Extra Panels')
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.options.split_extras:
+            excluded_items.add('Extra Panels')
+        else:
+            excluded_items.add('Extra 1')
+            excluded_items.add('Extra 2')
+            excluded_items.add('Extra 3')
+            excluded_items.add('Extra 4')
+            excluded_items.add('Extra 5')
+            excluded_items.add('Extra 6')
+
+        if self.options.split_bonus:
+            excluded_items.add('Bonus Panels')
+        else:
+            excluded_items.add('Bonus 1')
+            excluded_items.add('Bonus 2')
+            excluded_items.add('Bonus 3')
+            excluded_items.add('Bonus 4')
+            excluded_items.add('Bonus 5')
+            excluded_items.add('Bonus 6')
+
+        return excluded_items
+
+    def create_item_with_correct_settings(self, player: int, name: str) -> Item:
+        data = item_table[name]
+        item = Item(name, data.classification, data.code, player)
+
+        if not item.advancement:
+            return item
+
+        if name == 'Car Morph' and not self.options.stage_logic:
+            item.classification = ItemClassification.useful
+
+        if name == 'Secret Lens' and (self.options.hidden_object_visibility >= 2 or self.options.stage_logic != 0):
+            item.classification = ItemClassification.useful
+
+        if name in ["Bonus 1", "Bonus 2", "Bonus 3", "Bonus 4", "Bonus 5", "Bonus 6", "Bonus Panels"] and self.options.minigame_checks <= 1:
+            item.classification = ItemClassification.useful
+
+        if name in ["Bonus 1", "Bonus 3", "Bonus 4", 'Bonus Panels'] and self.options.item_logic == 1:
+            item.classification = ItemClassification.progression
+
+        if name == 'Piece of Luigi' and self.options.goal != 0:
+            if self.luigi_count >= self.options.luigi_pieces_required.value:
+                item.classification = ItemClassification.useful
+            else:
+                item.classification = ItemClassification.progression_skip_balancing
+                self.luigi_count += 1
+
+        return item
+
+    def generate_filler(self, multiworld: MultiWorld, player: int,
+                                        pool: List[Item]) -> None:
+        if self.playergoal == 1:
```

Comment:
This is another place to think about how someone reading the code will understand it.

It's not clear that `playergoal` is just the option from the player options.
It's not clear what `1` means, so it's not clear what this is checking, or why.

It could be:
```python
        if self.options.goal.value == PlayerGoal.option_luigi_hunt:
```